src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -187,28 +184,16 @@ public void storePlace(Place place) {
   /** Get all the places currently in the Datastore. */
   public List<Place> getAllPlaces() {
     List<Place> places = new ArrayList<>();
-    Query query = new Query("Place")
-        .addSort("timestamp", SortDirection.DESCENDING);
-
-    PreparedQuery results = datastore.prepare(query);
-    for (Entity entity : results.asIterable()) {
-      try {
-        String idString = entity.getKey().getName();
-        UUID id = UUID.fromString(idString);
-        String owner = (String) entity.getProperty("owner");
-        String title = (String) entity.getProperty("title");
-        String description = (String) entity.getProperty("description");
-        long latitude = (long) entity.getProperty("latitude");
-        long longitude = (long) entity.getProperty("longitude");
-        long timestamp = (long) entity.getProperty("timestamp");
-        Place place = new Place(id, owner, title, description, latitude, longitude, timestamp);
+    Set<String> users = getUsers();
+  
+    for (String userString : users) {
+      User user = getUser(userString);
```

Comment:
This is the N+1 query problem -- you're issuing one query, and then you're issuing N more queries to pull together the whole dataset. This is very inefficient and can cause n^2 scaling as the number of users grow.

https://www.google.com/search?q=n%2B1+query+problem&oq=N%2B1+query+problem&aqs=chrome.0.0l6.2824j0j4&sourceid=chrome&ie=UTF-8

The way I would prefer seeing you deduplicate this code if you kept the original code but moved the part which pulled values out of entity into another function. Then both `getAllPlaces` and `getPlaces` could call that one central function. 

A rule of thumb: if you find yourself repeating the same code in two functions and a reviewer asks you to deduplicate, the easiest fix is to pull out the common code into a third function, then have your original two functions call the third function. You should do this generally unless your reviewer specifically asks you to have one function call the other.

In this case, it makes sense to overload the constructor of Place so that it accepts an entity. You can see how I did this with User: https://github.com/jajanet/team-08-codeu/pull/39/files#diff-c40522f17c506878853f51f5851b4723R27

Reply:
okay, that makes sense! thanks!!